apps/calendar/js/views/modify_event.js

```diff
@@ -130,105 +128,26 @@ ModifyEvent.prototype = {
       return !!this.provider;
   },
 
-  /**
-   * Build the initial list of calendar ids.
-   */
   onfirstseen: function() {
-    // we need to notify users (specially automation tests) somehow that the
-    // options are still being loaded from DB, this is very important to
-    // avoid race conditions (eg.  trying to set calendar before list is
-    // built) notice that we also add the class to the markup because on some
-    // really rare occasions "onfirstseen" is called after the EventBase
-    // removed the "loading" class from the root element (seen it happen less
-    // than 1% of the time)
-    this.getEl('calendarId').classList.add(self.LOADING);
-
-    var calendarStore = this.app.store('Calendar');
-    calendarStore.all(function(err, calendars) {
-      if (err) {
-        return console.error('Could not build list of calendars');
-      }
-
-      var pending = 0;
-      var self = this;
-
-      function next() {
-        if (!--pending) {
-          self.getEl('calendarId').classList.remove(self.LOADING);
-
-          if (self.onafteronfirstseen) {
-            self.onafteronfirstseen();
-          }
-        }
-      }
-
-      for (var id in calendars) {
-        pending++;
-        this._addCalendarId(id, calendars[id], next);
-      }
-
-    }.bind(this));
+    if (this._loadingCalendarList) {
+      this.getEl('calendarId').classList.add(this.LOADING);
+    }
   },
 
-  /**
-   * Updates a calendar id option.
-   *
-   * @param {String} id calendar id.
-   * @param {Calendar.Model.Calendar} calendar model.
-   */
-  _updateCalendarId: function(id, calendar) {
+  _renderCalendarSelect: function(calendarList) {
     var element = this.getEl('calendarId');
-    var option = element.querySelector('[value="' + id + '"]');
-    var store = this.app.store('Calendar');
-
-    store.providerFor(calendar, function(err, provider) {
-      var caps = provider.calendarCapabilities(
-        calendar
-      );
-
-      if (!caps.canCreateEvent) {
-        this._removeCalendarId(id);
-        return;
-      }
-
-      if (option) {
-        option.text = calendar.remote.name;
-      }
-
-
-      if (this.oncalendarupdate) {
-        this.oncalendarupdate(calendar);
-      }
-    }.bind(this));
-  },
-
-  /**
-   * Add a single calendar id.
-   *
-   * @param {String} id calendar id.
-   * @param {Calendar.Model.Calendar} calendar calendar to add.
-   */
-  _addCalendarId: function(id, calendar, callback) {
-    var store = this.app.store('Calendar');
-    store.providerFor(calendar, function(err, provider) {
-      var caps = provider.calendarCapabilities(
-        calendar
-      );
-
-      if (!caps.canCreateEvent) {
-        if (callback) {
-          nextTick(callback);
-        }
+    element.innerHTML = '';
+    forEach(calendarList, (id, object) => {
+      var calendar = object.calendar;
+      var capabilities = object.capabilities;
+      if (!calendar.localDisplayed || !capabilities.canCreateEvent) {
```

Comment:
Only want to show the calendar as selectable if it's toggled visible and we have caldav permissions to write events to the dav collection